Dotwise - Analytics Website/Data Platform RFP

tolgayayci opened the pull request:
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [X] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [X] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [X] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [X] The software delivered for this grant will be released under an open-source license specified in the application.
- [X] The initial PR contains only one commit (squash and force-push if needed).
- [X] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

tolgayayci commented:
Hey @keeganquigley, I've created this proposal for [this RFP](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/analysis-website-and-data-platform.md) that you proposed. Excitedly waiting for your suggestions!

semuelle commented on applications/dotwise.md:
```diff
@@ -0,0 +1,191 @@
+
+  ![techstack](https://github.com/tolgayayci/Grants-Program/assets/40897846/21ed9c58-1950-4e8a-bf28-9e484a1eaf4b)
+
+  **Subquery:** Subquery is a tool for building custom data indexes for Polkadot, Kusama, and other Substrate-based networks. It is a great tool for building custom data indexes. It provides a great developer experience with many built-in features such as GraphQL support, TypeScript support, and Substrate support. It also has a great community and ecosystem with many plugins and tools.
```
Can you elaborate how this data index looks like? Do you just tell SubQuery to index everything and then any query I can think of is available? Or do complex queries require a customisation of this index?

tolgayayci replied:
I will use the subquery to index the customized data and then save it to my database, and for this I need to determine its scope and proceed in this way. Indexing and querying everything is difficult and not a very possible task for this project. Instead, I aim to create detailed indexes under core infrastructure such as nft details, account analysis, token analysis and so more.

Users can write their own queries on these indexed data, or they can create queries by simply selecting the attributes and related properties without writing a query with the module I will create.

semuelle reviewed (changes requested):
Thanks for the application, @tolgayayci! A couple of comments/questions:

tolgayayci commented on applications/dotwise.md:
```diff
@@ -0,0 +1,191 @@
+| 1. | SubQuery Integration | I will integrate SubQuery, into the backend of the project. This will allow users to search, filter, and aggregate data in a more efficient manner. Users just need to select attributes on UI to write a query with SubQuery via Core API. |
+| 2. | Database Integration | I will integrate MongoDB to store queries, visualizations, some stats and so more. |
+| 4. | Core API | The platform will offer an API that allows to gather, retrieve, and update project-related data. The use of FastAPI will help to ensure high performance and easy integration with other components. |
+| 5. | Analytics | I will integrate monitoring solutions to forecast backend system like Grafana. |
```
Of course, that means analyses of the backend system with grafana. This could be Core API analysis, Subquery usage analysis and so more. For example, Grafana firstly will be used to monitor subquery usage, if I monitor the system well, I can make my improvements better.

tolgayayci commented on applications/dotwise.md:
```diff
@@ -0,0 +1,191 @@
+
+- Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem?
+  
+  No, there isn't any similar project in the Polkadot ecosystem but there is developing solutions about data analysis as my knowledge. So if there will be any good data source ı can rich the my data sources and make the platform more useful.
```
I hadn't heard of the Web3Go project, I looked it up when you mentioned it, and it was first applied about two years ago, so I may have overlooked it. Although it is similar in proposal, as I have stated in my mock ups, when my project ends, it will have a working logic similar to Dune Analytics and I can say that it will be used in a simpler way.

Another difference is that Dotwise uses graphql and non-technical users can create queries by simply selecting attributes. Additionally, except for query and visualization, widgets like text, image can be created in the form of drag&drop on a dashboard.

tolgayayci commented on applications/dotwise.md:
```diff
@@ -0,0 +1,191 @@
+
+### Team's experience
+
+As a full-stack developer with 2 years of experience, I have honed my skills in software development, with a focus on dApp development in the past year. I have a deep interest in the Web3 and NFT space and have put my skills to the test by creating a number of relevant applications. In addition to my experience, I have developed detailed React and Next.js projects, further enhancing my ability to ...
```
I do not have a project that I have developed on data analysis on Substrate and I am making my first application on this subject. I have been interesting with Polkadot for a while and I aim to develop an effective data analysis application with this project. 

I decided to use SubQuery as the basis for this, because it helps me develop a system that will serve the purposes of the project quickly according to my research and experimentation. As I stated, after v1 completed I will consider to add other data sources as well.

tolgayayci commented:
Thanks for reply @semuelle 

It would be great to take help about scope of data indexing, definitely want to discuss!

I understand your concerns about the size of the project, instead of downgrading to level 1 only for subquery and graphql editor, I can add graphql editor to my first milestone delivery. When I develop only these two features, mostly the frontend development remains, which leaves the scope of the project incomplete. I'm curious about your opinions on this subject.

After the first milestone is successfully completed, the frontend part remains and I have experience in this, I have built Awesome Polka before that and I am currently doing performance optimizations, after that I will start with the marketing staff and make the announcements.

takahser reviewed (changes requested):
@tolgayayci could you add more info about the data that's being indexed? E.g. what data are you indexing, what's the data structure, how do you index them, etc. Apart from that, more details on the architecture and technologies that you're planning to implement/use would be useful as well. Finally, I'm not sure if solely supporting GraphQL rather than SQL is a good idea, since SQL is much more widespread which makes it more useful for this use case.

tolgayayci commented:
Hey @takahser, thanks for review

Upon your suggestion, I changed graphql with sql, for this I had to change the database as well.

In the data models section, I gave basic information about how I plan to index with subquery and added 4 data types that I intend to implement, I am open to extra suggestions about data. However, although I have determined the general data classes, I need to work on which attributes will be included in depth, so I will plan for the low level details lastly.

Additionally, for NFT data type I thought about take help from Kodadot, they have a indexer on Subquery, would be great to discuss.

I added a visual for a better understanding of the system, and finally, I added the technologies I plan to use in the frontend and backend.

Waiting for your evaluations!

Noc2 reviewed (changes requested):
Thanks for the application. But I have one big concern here: Your [previous project ](https://github.com/tolgayayci/awesome-polka-docs) already doesn't seem to be maintained, and when I, for example, click on [Protocols](https://awesomepolka.org/projects?dev_projects%5BhierarchicalMenu%5D%5Bcategories.lvl0%5D%5B0%5D=Protocols), I see only three projects out of hundreds. How do you plan to change this and why will this project be different?

Noc2 reviewed (changes requested):
Thanks for the reply. I won't approve the application, given the above reason. I just took another look at your previous project and I'm personally not convinced. But I will share it again with the rest of the team.

tolgayayci commented:
Hey @Noc2,

As I mentioned I was working on performance improvements, it's been a month since my last milestone approval and I was on leave for two weeks.

In this process, I have just completed my developments and, as you can see in the images below, I have brought the performance of the site to the required level.

| Optimized Version  | Old Version |
| ------------- | ------------- |
| ![NewPolka](https://github.com/w3f/Grants-Program/assets/40897846/cf924df7-4d09-4376-9963-c1f82a908363) | ![OldPolka](https://github.com/w3f/Grants-Program/assets/40897846/c16adfe8-b710-41db-9898-3f2bd11295c0) |

However, we had removed the admin panel during the discussion of the previous proposal, but I completed it in this time, I had to add it manually without the admin panel, and now I can automatically approve the applications, control the access to the project pages or the privileges of the users, thanks to this panel.

If someone from the team wants to be an admin, I'll be happy to add them.

| Pending Applications | Approved Applications & Users |
| ------------- | ------------- |
|  ![Screenshot 2023-07-04 at 15-05-07 Screenshot](https://github.com/w3f/Grants-Program/assets/40897846/599e9020-b7a5-4fee-be73-23d8471309d4) |  ![Screenshot 2023-07-04 at 15-04-56 Screenshot](https://github.com/w3f/Grants-Program/assets/40897846/35ee00d3-5549-47fb-988d-1bcfb17e8f3f) |

Finally, I updated my project index and added more than 500 projects to the system, although the projects are visible, there are no details because they must be added by the project teams due to the nature of the system, I will reach the project owners in the future and ask them to open their accounts.

You can have a look updated projects here: https://awesomepolka.org/projects

I completed all of these changes after the last milestone approval and as I mentioned above, I was on leave for 2 weeks of this period. Taking these into account, I think that I did a good job in the time passed, and I will soon start marketing by creating the accounts of the project teams. 

Thank you again for your evaluation and if you have any questions about these innovations, I would be happy to answer them, and I would like you to re-evaluate your comments about Awesome Polka and this application in the light of the information I have provided.

semuelle commented:
Thanks again for the application, @tolgayayci. However, the committee has decided not to support your application at this point. Some points of concern have already been raised in the discussion, so I won't repeat them here. Additionally, we have recently signed three grants addressing one or both of the data analysis RFPs, so we'll retire them for now. 

Best of luck going forward. Feel free to apply again in the future.
